client/executor/wasmtime/src/runtime.rs

```diff
@@ -497,16 +559,16 @@ where
 /// [`prepare_runtime_artifact`]. Otherwise, there is a risk of arbitrary code execution with all
 /// implications.
 ///
-/// It is ok though if the `compiled_artifact` was created by code of another version or with
+/// It is ok though if the compiled artifact was created by code of another version or with
```

Comment:
I think this has to be updated wrt new requirements that came with supplying a path. IIRC, the safety invariant says that the file should not be changed.

This should also be propagated to the Polkadot side, at least in a form of comments.